client/state/selectors/is-woocommerce-core-profiler-flow.ts

```diff
@@ -0,0 +1,20 @@
+import { get } from 'lodash';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import getInitialQueryArguments from 'calypso/state/selectors/get-initial-query-arguments';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Returns true if the user reached Calypso via the WooCommerce Core Profiler flow.
+ * This is indicated by the `from` query argument being set to `woocommerce-core-profiler`.
+ *
+ * @param  {Object}   state  Global state tree
+ * @returns {?boolean}        Whether the user reached Calypso via the WooCommerce Core Profiler flow
+ */
+export const isWooCommerceCoreProfilerFlow = ( state: AppState ): boolean => {
+	return (
+		'woocommerce-core-profiler' === get( getCurrentQueryArguments( state ), 'from' ) ||
+		'woocommerce-core-profiler' === get( getInitialQueryArguments( state ), 'from' )
```

Comment:
I'm curious about the addition of `getInitialQueryArguments` is added here. Doesn't seem to change the behaviour, but do we really need it? 🤔

Reply:
I had to add it for the 2fa screens because the url isn’t kept in the browser bar on those screens 😆 I figured it doesn’t harm any existing workflow to combine them because I can’t foresee a flow where we start off with core profiler flow and end up not being core profiler flow

Reply:
Agreed! Thanks for the context.